internal/sirius/get_refdata.go

```diff
@@ -0,0 +1,81 @@
+package sirius
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/ministryofjustice/opg-sirius-supervision-deputy-hub/internal/model"
+	"io"
+	"net/http"
+	"strings"
+)
+
+func (c *Client) GetRefData(ctx Context, refDataType string) ([]model.RefData, error) {
```

Comment:
What's the reason for this change? By passing in refDataType as a string, you've lost all type safety and removed the interfaces indicating what refdata is fetchable.

If you are going down this route, it would be much better to make GetRefData private (lower case) and export the functions for fetching the specific refdatas, which then just call this one with string values.

Reply:
The reason was to get rid of duplicated code across these refdata endpoints, but I hadn't thought about type safety - I'll give that a shot